analytics_test.go

```diff
@@ -0,0 +1,327 @@
+package analytics
+
+import (
+	"errors"
+	"testing"
+	"time"
+)
+
+// New should return a *Client with default value
+func TestNew(t *testing.T) {
+	cases := []struct {
+		key             string
+		expected_client *Client
+	}{
+		{
+			"examplekey",
+			&Client{
+				FlushAt:    20,
+				FlushAfter: 5 * time.Second,
+				Key:        "examplekey",
+				Endpoint:   "https://api.segment.io",
+			},
+		},
+	}
+
+	for _, c := range cases {
+		got := New(c.key)
+		if got.FlushAt != c.expected_client.FlushAt {
+			t.Errorf("got: %v, expected: %v", got.FlushAt, c.expected_client.FlushAt)
+		}
+		if got.FlushAfter != c.expected_client.FlushAfter {
+			t.Errorf("got: %v, expected: %v", got.FlushAfter, c.expected_client.FlushAfter)
+		}
+		if got.Key != c.expected_client.Key {
+			t.Errorf("got: %v, expected: %v", got.Key, c.expected_client.Key)
+		}
+		if got.Endpoint != c.expected_client.Endpoint {
+			t.Errorf("got: %v, expected: %v", got.Endpoint, c.expected_client.Endpoint)
+		}
+	}
+}
+
+// Identify should return an expected error if "userId" or "anonymousId" doesn't exists
+func TestIdentify(t *testing.T) {
+	valid_client := New("this is a secret, you can't read it. don't you?")
+	cases := []struct {
+		client       *Client
```

Comment:
I don't see the client field used referenced anywhere. Let's remove it if it's unused.